src/strategies/saffron-finance/index.ts

```diff
@@ -0,0 +1,273 @@
+  const dexReserveData = new Array<DexReserveSupply>();
+  options.dexLpTypes.forEach(dexToken => {
+    const d:DexReserveSupply = {
+      name: dexToken.name,
+      reservesQuery: [dexToken.lpToken, 'getReserves'],
+      reserveQueryIdx: 0,
+      reserve: BigNumber.from(0),
+      supplyQuery: [dexToken.lpToken, 'totalSupply'],
+      supplyQueryIdx: 0,
+      supply: BigNumber.from(0),
+      saffLpToSFI_E18: BigNumber.from(0)
+    }
+    callQueries.push(d.reservesQuery);
+    d.reserveQueryIdx = callQueryIndex++;
+    callQueries.push(d.supplyQuery);
+    d.supplyQueryIdx = callQueryIndex++;
+    dexReserveData.push(d);
+
+    const batch: Batch = {tag: `${dexToken.name}`, votingScheme: "", qIdxStart: dexLpCallIdxStart, qIdxEnd: callQueryIndex};
+    dexLpPairQueryBatches.push(batch);
+    dexLpCallIdxStart = callQueryIndex;
+  });
+
+  // ============= Multicall queries ==============
+  options.contracts.forEach(contract => {
+    const queries = addresses.map((address: any) => {
+      return [ contract.tokenAddress, 'balanceOf', [address] ];
+    });
+    const queriesLength = callQueries.push(...queries);
+    const batch = {tag: contract.label, votingScheme: contract.votingScheme, qIdxStart: callQueryIndex, qIdxEnd: queriesLength};
+    callQueryIndex = queriesLength;
+    holdersQueryBatches.push(batch);
+  });
+
+  // Run queries
+  callResponses = await multicall(network, provider, abi, callQueries, {blockTag});
+
+  // ========== Extract and process query responses ==========
+  dexReserveData.forEach(drd => {
+    drd.reserve = callResponses[drd.reserveQueryIdx][0];
+    drd.supply = callResponses[drd.supplyQueryIdx][0];
+    drd.saffLpToSFI_E18 = drd.reserve.mul(BIG18).div(drd.supply);
+  });
+
+
+  // ========== Build the voting schemes and calculate indiviudal scores ============
+  const voteScorer: VoteScorer = new VoteScorer(dexReserveData);
+  options.votingSchemes.forEach(scheme => {
+    voteScorer.createVotingScheme(scheme.name, scheme.type, scheme.multiplier);
+  });
+
+  options.contracts.forEach(contract => {
+    const batch = holdersQueryBatches.find(e => e.tag === contract.label);
+    if (batch === undefined) {
+      throw new Error(`Failed to locate tag, ${contract.label}, in queryBatches.`);
+    }
+    let idxStart = batch.qIdxStart;
+    const batchScores = addresses.map((address: any, index: number) => {
+      return {address: address, score: voteScorer.calculateScore(contract.votingScheme, callResponses[idxStart+index][0])};
+    });
+    votingScores.push(...batchScores);
+  });
+
+  // ================ Sum up everything =================
+  let addressVotingScore = addresses.map((address: any, index: number) => {
+    let total = BigNumber.from(0);
+    holdersQueryBatches.forEach((batch: Batch, index: number) => {
+      let votingScore = votingScores[batch.qIdxStart+index];
+      if (votingScore.address === address) {
```

Comment:
@turpintinz Just tried the strategy, it fail for me at this line with `TypeError: Cannot read property 'address' of undefined`.

Reply:
Yep, me too. I'll be repairing that today, April 9. Thanks for asking.

Reply:
> @turpintinz Just tried the strategy, it fail for me at this line with `TypeError: Cannot read property 'address' of undefined`.

This is now repaired.